app/controllers/topics_controller.rb

```diff
@@ -842,6 +842,7 @@ def timings
   def feed
     @topic_view = TopicView.new(params[:topic_id])
     discourse_expires_in 1.minute
+    raise Discourse::NotFound if @topic_view.posts.blank?
```

Comment:
Loading the whole `TopicView` might be "heavy" one the db. I wonder if there's an easy way to make a tiny check that there's at least one post before loading the `TopicView`?